### ISSUE TITLE
Recipients not syncing after secret expired

### ISSUE THREAD

**Issue by ostromecko** [cristianoag/microsoft-teams-apps-company-communicator#76]
Hi, I'm running the latest master version v5.31. Recently our secrets have expired, and what we did is that we generated new ones, and update them in key vault. Currently I'm receiving below error when trying to send a test message using a CSV file:
![image](https://github.com/cristianoag/microsoft-teams-apps-company-communicator/assets/134921401/a5ea170b-efcc-42df-92e8-ce55921be686)

Failed to prepare the message for sending:The orchestrator function 'SyncRecipientsOrchestrator' failed: "The activity function 'SyncCSVActivity' failed: "Specified argument was out of the range of valid values. (Parameter 'partitionCount') Actual value was .". See the function execution logs for additional details.". See the function execution logs for additional details.

So I would need help with resolving this, and checking if I did not miss any secret update anywhere in the configuration.

**Comment by luishdemetrio** [comment]
Hi @ostromecko, could you please share the function execution logs?

The Log Stream is a feature in Azure that allows you to view the live logging information directly in the Azure portal. Here’s how you can use it:

1. Navigate to your function app in the Azure portal.
2. Select the Monitoring section.
3. Click on Log Stream.

Once you’re in the Log Stream, you should start seeing logs as your functions execute. If you don’t see any logs, make sure your function app is running and executing functions.

**Comment by ostromecko** [comment]
Hi @luishdemetrio, which function from the three functions I should open and check if there are logs? Data-Function, Function or Prep-function? 
And what you mean by "make sure your function app is running and executing functions."? Company communicator is an app which is used ad-hoc.

**Comment by ostromecko** [comment]
Hi @luishdemetrio, can you please support me with the check if the functions are executed?

**Comment by ostromecko** [comment]
> Hi @ostromecko, could you please share the function execution logs?
> 
> The Log Stream is a feature in Azure that allows you to view the live logging information directly in the Azure portal. Here’s how you can use it:
> 
> 1. Navigate to your function app in the Azure portal.
> 2. Select the Monitoring section.
> 3. Click on Log Stream.
> 
> Once you’re in the Log Stream, you should start seeing logs as your functions execute. If you don’t see any logs, make sure your function app is running and executing functions.

@luishdemetrio I have opened Log stream, but it's constantly loading and not showing anything. Can you please support on this?

**Comment by ostromecko** [comment]
@cristianoag could you help me with this issue?

**Comment by cristianoag** [comment]
When you renew the secrets in KeyVault you need to make sure the configuration variables are referencing the new key. Can you check all functions and the app service and make sure they are pointing to the right place?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ostromecko** [comment]
Currently I'm facing this issue:
Failed to prepare the message for sending:The orchestrator function 'SyncRecipientsOrchestrator' failed: "The activity function 'SyncCSVActivity' failed: "Specified argument was out of the range of valid values. (Parameter 'partitionCount') Actual value was .". See the function execution logs for additional details.". See the function execution logs for additional details.

**Comment by cristianoag** [comment]
Are you using a standard CSV file as input? Can you post it so I can check the format?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ostromecko** [comment]
of course. I'm attaching file which I'm using. 
[companycomm.csv](https://github.com/cristianoag/microsoft-teams-apps-company-communicator/files/14723177/companycomm.csv)

**Comment by cristianoag** [comment]
I've just tested, and the format you provided appears to be in order and is being recognized. I suggest considering a few potential areas for investigation.

Verify if all functions are correctly configured to access the appropriate keys in KeyVault. Assuming you've already done this, another possibility could be discrepancies between the versions of functions being run compared to the version of the app service. It's essential to ensure that all resources are synchronized to the same version. I recommend checking the synchronization log file for any errors or inconsistencies.

**Comment by ostromecko** [comment]
I have checked all three functions and app service. All four instances are pointing to the same commit ID 9d9ce88.

**Comment by cristianoag** [comment]
Do you have email addresses different than upns?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ostromecko** [comment]
No. I was even testing this with a file which was containing only two email address (mine, and my colleague) and still it was not working

**Comment by cristianoag** [comment]
Email address or UPN? Very important, the file needs to have upns, not emails.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ostromecko** [comment]
UPN matches e-mail address in our case.

**Comment by cristianoag** [comment]
Collect and upload the function log files. So we can check what is going on. Need more detailed error mess

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ostromecko** [comment]
Unfortunatly the log stream for me is constantly loading, and it doesn't provide any logs 
![image](https://github.com/cristianoag/microsoft-teams-apps-company-communicator/assets/134921401/e99a3078-02d0-4b9a-a919-1ee001daefc3)

Or I can collect them in different way?

**Comment by onealll** [comment]
@cristianoag Im facing the same problem, all my secrets has expired, you have any script to do the secrets update to the keyvault?
or can you tell me what to do exacly to update?, im using versing 5.31
![expired](https://github.com/user-attachments/assets/1473513c-6cd1-4829-86b0-dbb15a739736)
The problem is this, when I try to select the users to send the message, Im reciving this messagem:
![imagem](https://github.com/user-attachments/assets/42453e89-7ca3-4873-b09c-a2f6d544fd5a)

"Permissions required. Ask your IT administrator to enable this option for you"

Im assuming this message is becouse of the expired secredts that don allow access to groups/users of m365
